src/js/components/picklist/index.js

```diff
@@ -59,7 +59,7 @@ const PicklistItem = View.extend({
   tagName: 'li',
   itemTemplate: hbs`
     <div class="flex-grow">{{#if icon}}{{fa icon.type icon.icon classes=icon.classes}}{{/if}}<span>{{matchText text query}}</span></div>
-    <div>{{#if isChecked}}{{fas "circle-check" classes="u-icon--12 u-margin--l-16"}}{{/if}}</div>
+    {{#if isChecked}}{{fas "check" classes="u-icon--12 u-margin--l-16"}}{{/if}}
```

Comment:
As you stated on Shortcut, the wrapping `<div>` stopped the `align-items: center` from working. Removing that fixed the vertical alignment issue with the checkmark.

And I also switched the icon for this from `circle-check` to `check`. As requested by Mark.